packages/context/src/binding-filter.ts

```diff
@@ -9,9 +9,33 @@ import {BindingAddress} from './binding-key';
 /**
  * A function that filters bindings. It returns `true` to select a given
  * binding.
+ *
+ * **NOTE**: Originally, we allow filters to be tied with a single value type.
+ * This actually does not make much sense - the filter function is typically
+ * invoked on all bindings to find those ones matching the given criteria.
+ * Filters must be prepared to handle bindings of any value type. We learned
+ * about this problem after enabling TypeScript's `strictFunctionTypes` check,
+ * but decided to preserve `ValueType` argument for backwards compatibility.
```

Comment:
Can you add TODO(semver-major) comment please, to make it easy for us to find breaking changes to include when making a new semver-major release.